Feature/79549 rsvp acceptance

niconerd opened the pull request:
MR ticket: https://central.tri.be/issues/106603
Original research ticket: https://central.tri.be/issues/79549

ggwicz commented on tests/_support/AcceptanceTester.php:
```diff
@@ -23,4 +23,39 @@ class AcceptanceTester extends \Codeception\Actor
+}
+PHP;
+
+		$I->haveMuPlugin('ticketable-post-types.php',$code);*/
```
any reason to keep these comments in place? if not, remove

niconerd replied:
@lucatume I found the DB method slighly faster while running the tests but I'd defer to your opinion here.

lucatume replied:
If they proved to be faster then use the method you found best and remove this code, thanks!

niconerd replied:
Actually re-tested and it's not. I'll change back to the muplugin method.

ggwicz commented on tests/_support/AcceptanceTester.php:
```diff
@@ -23,4 +23,39 @@ class AcceptanceTester extends \Codeception\Actor
+	 */
+	public function haveTicketablePostTypes ( array $post_types = array( ) ) {
+
+		$I = $this;
```
you don't _need_ to change this, but I'd recommend just using `$this` all of the time instead of aliasing it to `$I`

although if this is a common (or even recommended) pattern with acceptance tests, which it seems to be, then please ignore this comment 😎

niconerd replied:
@ggwicz this seems to be the good practice, found it this way in the official codeception docs.

ggwicz commented on tests/_support/AcceptanceTester.php:
```diff
@@ -23,4 +23,39 @@ class AcceptanceTester extends \Codeception\Actor
+	 *
+	 * @param  array $post_types The post types to be enabled for tickets. Can be empty.
+	 */
+	public function haveTicketablePostTypes ( array $post_types = array( ) ) {
```
our method/function/var names should always be `snake_case` in PHP

niconerd replied:
@ggwicz in this case I'm following the casing of the 'standard' of the existing Actor methods and the ones added by WP Browser. Not sure if my approach is the correct one.

lucatume replied:
Both ways are really fine. My personal way is: anything I add in the context of WordPress related work I `snake_case`.

ggwicz commented on tests/_support/AcceptanceTester.php:
```diff
@@ -23,4 +23,39 @@ class AcceptanceTester extends \Codeception\Actor
+	/**
+	 * Activates The Events Calendar
+	 */
+	public function haveTheEventsCalendarActive ( ) {
```
`snake_case`

ggwicz commented on tests/acceptance/Commerce/RSVP/TicketCreationCest.php:
```diff
@@ -0,0 +1,83 @@
+	 *
+     * @return array
+     */
+	protected function post_types_provider () {
```
fix  php docblock formatting here

niconerd commented:
@ggwicz thanks a lot for the review. I added some comments on specific things and fixed the other stuff (commit incoming). Regarding the comments these tests will serve as example for other QA folks digging into this, that's why there are almost obvious comments. But if you think this should be documented in other fashion anyway, please let me know.

niconerd commented:
@ggwicz can you take a new look? I decided to keept the `$I` and the camel casing for this cases.

@lucatume I see an integration test failing in the CI. Can you help me fix that one? Also seems like the job time is exceeded. I took a look at the code but it far above my technical chops. Thanks.

bordoni commented:
We need to resolve these conflicts, after that I feel like this is good for a merge.

niconerd commented:
@bordoni I think there's more work to be done in the setup. Right now running the tests as they are (with current acceptance setup) would produce failures.

Camwyn commented:
A lot has changed since the project was started, and things interfered with this being finished prior - now the environment is vastly different (blocks!) so this PR (and the associated ticket) is on hold until we determine the direction we're going with it.

sc0ttkclark commented:
Going to close this PR because we've got a lot to revisit on this. We can open up a new PR whenever we have time to resolve the problems.

sc0ttkclark commented:
Related: https://central.tri.be/issues/106603
